### PR TITLE
Support clojure-ts-mode 

### LINKED ISSUE #25: Support clojure-ts-mode out of the box
It would be nice if next to `clojure-mode` (and its cljs and cljc variants) also [`clojure-ts-mode`](https://github.com/clojure-emacs/clojure-ts-mode) and its variants were supported by flycheck-clj-kondo out of the box.

Meanwhile, in my emacs config, I enable with an adapted `flycheck-clj-kondo-define-checkers` macro.

```elisp
(defmacro flycheck-clj-kondo-define-ts-checkers (&rest extra-args)
  "Defines all clj-kondo checkers.
Argument EXTRA-ARGS: passes extra arguments to the checkers."
  `(progn
     (flycheck-clj-kondo--define-checker clj-kondo-clj-ts "clj" clojure-ts-mode ,@extra-args)
     (flycheck-clj-kondo--define-checker clj-kondo-cljs-ts "cljs" clojurescript-ts-mode ,@extra-args)
     (flycheck-clj-kondo--define-checker clj-kondo-cljc-ts "cljc" clojurec-ts-mode ,@extra-args)
     (flycheck-clj-kondo--define-checker clj-kondo-edn-ts "edn" clojure-ts-mode ,@extra-args)
     (dolist (element '(clj-kondo-clj-ts clj-kondo-cljs-ts clj-kondo-cljc-ts clj-kondo-edn-ts))
       (add-to-list 'flycheck-checkers element))))

(flycheck-clj-kondo-define-ts-checkers "--cache")
```

### PR DESCRIPTION
**Pull request by kommen**
Closes #25

@borkdude how do you want to handle the tests, as clojure-ts-mode yet is trivial to set up everywhere?

### CODE REVIEW

**Comment by borkdude** [comment]
There are some tests here: https://github.com/borkdude/flycheck-clj-kondo/blob/master/test/flycheck-clj-kondo-test.el

File: flycheck-clj-kondo.el [hunk 1]
```diff
@@ -53,7 +53,7 @@
   your file extension doesn't match your major-mode.")
 
 (defmacro flycheck-clj-kondo--define-checker
-    (name lang mode &rest extra-args)
+    (name lang modes &rest extra-args)
```
Comment (borkdude) [inline flycheck-clj-kondo.el:56]: Isn't this a breaking change?

Comment (kommen) [reply flycheck-clj-kondo.el:56]: With https://github.com/borkdude/flycheck-clj-kondo/pull/26/commits/368bbb026a7346cde10efdfd60abb1637d5d95b7 it is no longer a breaking change as per the flycheck docs `:modes` can be either a symbol ofter major mode or a list of symbols of major modes.

But you were right, my previous changes to `:predicate` would have been breaking.

File: flycheck-clj-kondo.el [hunk 2]
```diff
@@ -86,18 +86,20 @@ Argument EXTRA-ARGS: passes extra args to the checker."
 
                    ;; Else use the mode to infer which language to turn on.
                    (pcase ,lang
-                     ("clj" `(equal 'clojure-mode major-mode))
-                     ("cljs" `(equal 'clojurescript-mode major-mode))
-                     ("cljc" `(equal 'clojurec-mode major-mode)))))))
+                     ("clj"  (member major-mode '(clojure-mode clojure-ts-mode)))
+                     ("cljs" (member major-mode '(clojurescript-mode clojurescritps-ts-mode)))
+                     ("cljc" (member major-mode '(clojurec-mode clojurec-ts-mode))))))))
+
```
Comment (kommen) [inline flycheck-clj-kondo.el:92]: Note, I don't fully understand what this code intended to do. It seems it is doing the same as the `:modes` argument?

Comment (borkdude) [reply flycheck-clj-kondo.el:92]: I don't know

File: flycheck-clj-kondo.el [hunk 3]
```diff
@@ -86,18 +86,20 @@ Argument EXTRA-ARGS: passes extra args to the checker."
...
 (defmacro flycheck-clj-kondo-define-checkers (&rest extra-args)
   "Defines all clj-kondo checkers.
 Argument EXTRA-ARGS: passes extra arguments to the checkers."
   `(progn
-     (flycheck-clj-kondo--define-checker clj-kondo-clj "clj" clojure-mode ,@extra-args)
-     (flycheck-clj-kondo--define-checker clj-kondo-cljs "cljs" clojurescript-mode ,@extra-args)
-     (flycheck-clj-kondo--define-checker clj-kondo-cljc "cljc" clojurec-mode ,@extra-args)
-     (flycheck-clj-kondo--define-checker clj-kondo-edn "edn" clojure-mode ,@extra-args)
+     (flycheck-clj-kondo--define-checker clj-kondo-clj "clj" (clojure-mode clojure-ts-mode) ,@extra-args)
+     (flycheck-clj-kondo--define-checker clj-kondo-cljs "cljs" (clojurescript-mode clojurescritps-ts-mode) ,@extra-args)
```
Comment (borkdude) [inline flycheck-clj-kondo.el]: there seems to be a typo

**Comment by borkdude** [comment]
Thanks a lot.

**Pull request merged**